src/pyconverter/xml2py/writer.py

```diff
@@ -295,90 +391,246 @@ def write_source(
-        with open(path, "w", encoding="utf-8") as fid:
-            fid.write(cmd_obj.to_python(cmd_map, custom_functions))
+    if not os.path.isdir(library_path):
+        os.makedirs(library_path)
 
-        try:
-            exec(cmd_obj.to_python(cmd_map, custom_functions))
-        except:
-            raise RuntimeError(f"Failed to execute {cmd_name}.py") from None
+    module_name_list = []
 
-    mod_file = os.path.join(cmd_path, "__init__.py")
-    with open(mod_file, "w") as fid:
-        for ans_name in commands:
-            if ans_name in SKIP_XML:
+    if structured == False:
+        package_structure = None
+        for initial_command_name, command_obj in tqdm(command_map.items(), desc="Writing commands"):
+            if initial_command_name in SKIP_XML:
                 continue
-            cmd_name = cmd_map[ans_name]
-            fid.write(f"from .{cmd_name} import *\n")
-        fid.write("try:\n")
-        fid.write("    import importlib.metadata as importlib_metadata\n")
-        fid.write("except ModuleNotFoundError:\n")
-        fid.write("    import importlib_metadata\n\n")
-        fid.write("__version__ = importlib_metadata.version(__name__.replace('.', '-'))\n")
-        fid.write('"""PyConverter-GeneratedCommands version."""\n')
+            python_name = name_map[initial_command_name]
+            module_name_list.append(python_name)
+            path = os.path.join(library_path, f"{python_name}.py")
+            with open(path, "w", encoding="utf-8") as fid:
+                python_method = command_obj.to_python(custom_functions)
+                fid.write(f"{python_method}\n")
+            fid.close()
+
+            try:
+                exec(command_obj.to_python(custom_functions))
+            except:
+                raise RuntimeError(f"Failed to execute {python_name}.py") from None
+
+    else:
+        import subprocess
+
+        package_structure = {}
+        all_commands = []
+        specific_classes = get_config_data_value(config_path, "specific_classes")
+
+        for command in command_map.values():
+            if command.name in SKIP_XML:
+                continue
+            if command.group is None:
+                continue
+            initial_module_name, initial_class_name = command.group
+            initial_module_name = initial_module_name.replace("/", "")
+            module_name = initial_module_name.replace(" ", "_").lower()
+            module_path = os.path.join(library_path, module_name)
+            if not os.path.isdir(module_path):
+                os.makedirs(module_path)
+                package_structure[module_name] = {}
+            if initial_class_name in specific_classes.keys():
+                initial_class_name = specific_classes[initial_class_name]
+            class_name = initial_class_name.title().replace(" ", "").replace("/", "")
+            file_name = initial_class_name.replace(" ", "_").replace("/", "_").lower()
```

Comment:
You might want to add some empty lines between these statements (better readability) and group some statements into functions.